### ISSUE TITLE
Errors should update if onSubmit prevents default

### ISSUE THREAD

**Issue by penx** [edmundhung/conform#549]
### Describe the bug and the expected behavior

When calling preventDefault in onSubmit, form validation errors should be updated to the last call to onValidate on the next render.

Possibly related to #396.

### Conform version

v1.0.6

### Steps to Reproduce the Bug or Issue

https://stackblitz.com/edit/vitejs-vite-tqtrht

e.g. given the following code:

```tsx
import { parseWithZod } from '@conform-to/zod';
import { getFormProps, useForm } from '@conform-to/react';
import { z } from 'zod';
import { useState } from 'react';

const schema = z.object({
  test: z.string().min(1),
});

export default function App() {
  const [count, setCount] = useState(0);

  const [form, fields] = useForm({
    shouldValidate: 'onSubmit',
    onValidate({ formData }) {
      return parseWithZod(formData, { schema });
    },
    onSubmit: (e) => {
      e.preventDefault();
      // trigger a rerender - we're not submitting or navigating so no rerender happens
      setCount((val) => val + 1);
      return null;
    },
  });

  return (
    <form {...getFormProps(form)}>
      <input type="text" name="test" />
      <button type="submit">Submit {count}</button>
      <pre>{JSON.stringify(fields.test.errors, null, 2)}</pre>
    </form>
  );
}
```

If you:

- submit without entering any text
- notice validation error
- enter some text and submit again

Expected

- validation message should disappear

Actual:

- outdated validation message is still shown

### Additional context

For context, the reason I am doing this is because I'm building a component and showing, in storybook, that it is compatible with conform. I don't actually want the form to submit anywhere.

**Comment by edmundhung** [comment]
Form errors will be cleared immediately on form submit now instead of waiting until the server result is back with [v1.1.0-pre.0](https://github.com/edmundhung/conform/releases/tag/v1.1.0-pre.0).

**Comment by edmundhung** [comment]
This is fixed in [v1.1.0](https://github.com/edmundhung/conform/releases/tag/v1.1.0).
